src/html_parser.ml

```diff
@@ -2235,7 +2235,7 @@ let parse ?depth_limit requested_context report (tokens, set_tokenizer_state, se
       in_body_mode_rules "table" mode v
 
     | v ->
-      anything_else_in_table mode v
+      anything_else_in_table in_body_mode v
```

Comment:
It seems this commit doesn't achieve its intended purpose. Prior to this change, the parser was already switching to "in-body" mode inside the `anything_else_in_table` function:

https://github.com/ahrefs/markup.ml/blob/28bafacb824ebc8350735a41842448e7d8811a6e/src/html_parser.ml#L2157-L2159

I'm guessing this `mode` is the "original insertion mode": https://html.spec.whatwg.org/multipage/parsing.html#original-insertion-mode; and setting this explicitly to `in_body_mode` might have unintended consequences, and might have caused the regression bug.

Reply:
> `in_body_mode_rules "table" mode v)`

There my change puts `in_body_mode` in the `mode` instead of what was in there which is context dependent; in the case of the OOM bug it was staying in a (malformed?) `in_table_mode`.